src/jvm/test-jdk11/org/jetbrains/kotlinx/lincheck_test/representation/RunWithLambdaRepresentationTests.kt

```diff
@@ -0,0 +1,154 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import org.jetbrains.kotlinx.lincheck.Lincheck
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicIntegerArray
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicLongArray
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.atomic.AtomicReferenceArray
+import kotlin.random.Random
+
+
+abstract class BaseRunWithLambdaRepresentationTest(private val outputFileName: String) {
+    /**
+     * Implement me and place the logic to check its trace.
+     */
+    abstract fun block()
+
+    @Test
+    fun testRunWithModelChecker() {
+        val failure = Lincheck.verifyWithModelChecker {
+            block()
+        }
+        failure.checkLincheckOutput(outputFileName)
+    }
+}
```

Comment:
There are also tests in `CustomThreadsRepresentationTest.kt` file that I have added when I was working on supporting custom threads. Could you please move them in this file, rewrite them to use `verifyWithModelChecker` function and check that they would still work?